modules/src/core/ics03_connection/handler/conn_open_init.rs

```diff
@@ -73,24 +86,41 @@ mod tests {
             name: String,
             ctx: MockContext,
             msg: ConnectionMsg,
+            expected_versions: Vec<Version>,
             want_pass: bool,
         }
 
-        let msg_conn_init =
+        let msg_conn_init_default =
             MsgConnectionOpenInit::try_from(get_dummy_raw_msg_conn_open_init()).unwrap();
-        let context = MockContext::default();
+        let msg_conn_init_no_version = MsgConnectionOpenInit {
+            version: None,
+            ..msg_conn_init_default.clone()
+        };
+        let default_context = MockContext::default();
+        let good_context = default_context
+            .clone()
+            .with_client(&msg_conn_init_default.client_id, Height::new(0, 10));
 
         let tests: Vec<Test> = vec![
             Test {
                 name: "Processing fails because no client exists in the context".to_string(),
-                ctx: context.clone(),
-                msg: ConnectionMsg::ConnectionOpenInit(msg_conn_init.clone()),
+                ctx: default_context,
+                msg: ConnectionMsg::ConnectionOpenInit(msg_conn_init_default.clone()),
+                expected_versions: vec![msg_conn_init_default.version.clone().unwrap()],
                 want_pass: false,
             },
+            Test {
+                name: "No version in MsgConnectionOpenInit msg".to_string(),
```

Comment:
Can we add a test for when there is a version mismatch?

Reply:
Done